### ISSUE TITLE
[terra-date-picker] Accessibility Issues

### ISSUE THREAD

**Issue by benbcai** [cerner/terra-framework#1158]
# Bug Report

## Description
The date picker has several accessibility issues that should be addressed or enhanced to increase the accessibility of the component.

1. A date in the input is being read as text, not as a date. For example, an input date of `01/15/2020` is being read as "zero one slash fifteen slash twenty twenty". It should be read as "January fifteenth twenty twenty".
2. Visual focus should NOT be initially placed on the calendar. Instead, place focus at the very beginning of the popup so the user can tab through all controls or begin reading from the start. For any modal or popup, the focus should be on the actual container. There may not be a visual focus, but it would start the user at the very beginning of the content. A screen reader would announce the title of the modal or provide context to the popup and what it contains.
3. When navigating within the calendar, provide the day of week in programmatic context for each day. For example, a preselected day of 01/15/2020 is being ready as "January fifteenth twenty twenty". It should be enhanced to read "Wednesday January fifteenth twenty twenty".
4. The accessible name (aria-label) must match the visual label (`label` prop in DatePickerField). If the visual label is “Start Date” then the accessible name must also be “Start Date” – a mismatch in visual and accessible name would prevent users from activating the controls.
5. On open of the date picker calendar, nothing is announced to the user. User must be provided with context that the calendar is opened. Ideally, the user would be provided with context of what month/year the calendar is defaulted to and what, if any date is currently selected. Generally, the user does something, the system reacts. The system must ensure programmatic change of context and any new value is available to announce.
6. When selecting a date or clicking on the Today button, the picker closes and updates the date in the input but provides no programmatic notification of change of context.
 that a date has been selected and the picker is closed as a result. A better experience recommended by UX on clicking of the Today button would be to move user to today’s date in the date picker calendar and allow the user to select the date. However, this would be a breaking change. If the current behavior is kept, user must be provided context of calendar closing and the date that is selected. Perhaps the new date updated in the input is not announced because the focus has programmatically shifted to the next control to guide the workflow.
7. The color of the Previous and Next month button icons and their background colors have insufficient color contrast. 

## Steps to Reproduce

1. Turn on Voice Over. Go to the various [examples](https://engineering.cerner.com/terra-framework/components/terra-date-picker/date-picker/date-picker) and listen to what is announced.

## Environment

* Component Name and Version:
* Browser Name and Version:
* Node/npm Version: [e.g. Node 8/npm 5]
* Webpack Version:
* Operating System and version (desktop or mobile):
